app/code/Magento/Checkout/Block/Cart/Shipping.php

```diff
@@ -43,14 +51,17 @@ public function __construct(
         \Magento\Checkout\Model\CompositeConfigProvider $configProvider,
         array $layoutProcessors = [],
         array $data = [],
-        \Magento\Framework\Serialize\Serializer\Json $serializer = null
+        \Magento\Framework\Serialize\Serializer\Json $serializer = null,
+        \Magento\Checkout\Model\Cart\ConfigProvider $cartConfig = null
```

Comment:
Please inject this class via block argument in layout

Reply:
Hi @sivaschenko 

Based on the discussion in https://github.com/magento/magento2/pull/23733 I think it's not backwards compatible to add block arguments via layout xml?
If custom modules try to instantiate the same block class in some custom xml file, then after upgrading Magento the module might crash as it's missing some arguments which are then undefined but still being used by the phtml file.

Can you double check this?